applications/GeoMechanicsApplication/python_scripts/geomechanics_solver.py

```diff
@@ -137,6 +137,26 @@ def GetDefaultParameters(cls):
         this_defaults.AddMissingParameters(super().GetDefaultParameters())
         return this_defaults
 
+    def ValidateSettings(self):
+        """This function validates the settings of the solver
+        """
+
+        super().ValidateSettings()
+
+        # Checks if scaling is used in combination with rebuild level lower than 2 and prebuild dynamics, if so it
+        # throws an error
+        if (self.settings.Has("linear_solver_settings") and
+            self.settings["linear_solver_settings"].Has("scaling") and
+            self.settings["linear_solver_settings"]["scaling"].GetBool()):
+            if (self.settings.Has("rebuild_level") and
+                self.settings["rebuild_level"].GetInt() < 2):
+                raise ValueError("Scaling can only be used if rebuild level is 2")
+            if (self.settings.Has("prebuild_dynamics" and
```

Comment:
I think a parenthesis went missing:
```suggestion
            if (self.settings.Has("prebuild_dynamics") and
```